crates/autopilot/src/run_loop.rs

```diff
@@ -446,16 +446,8 @@ impl RunLoop {
         {
             futures::future::Either::Left((res, _)) => res,
             futures::future::Either::Right((driver_result, onchain_task)) => {
-                let driver_tx_hash = driver_result.map_err(SettleError::Failure)?.tx_hash;
-                let onchain_tx_hash = onchain_task.await?;
-                if onchain_tx_hash != driver_tx_hash {
-                    tracing::warn!(
-                        ?driver_tx_hash,
-                        ?onchain_tx_hash,
-                        "driver and onchain tx hashes do not match"
-                    );
-                }
-                Ok(onchain_tx_hash)
+                driver_result.map_err(SettleError::Failure)?;
+                onchain_task.await
```

Comment:
Now that a solver can only signal abort we no longer have to await the autopilot loop to compare transaction hashes. Technically a solver could still play games by pretending it aborted submitting while submitting anyways but the disruption caused by that would be very minor. However, we should still be able to track these cases to inform/disable solvers that repeatedly violate this so I think this should be logged here.

Reply:
But we are still awaiting the autopilot deadline despite the solver already telling us it gave up.

Reply:
But there is a question mark operator which should early return in case of error.